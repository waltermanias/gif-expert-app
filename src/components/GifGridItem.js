import React from "react";
import PropTypes from "prop-types";

const GifGridItem = ({ title, id, url }) => {
  return (
    <div className="card animate__animated animate__bounce animate__delay-2s">
      <img src={url} alt={"imagen"} />
      <p>{title}</p>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifGridItem;
