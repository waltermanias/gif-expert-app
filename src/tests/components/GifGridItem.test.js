import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("GifGridItem", () => {
  const wrapper = shallow(
    <GifGridItem title="My Title" url="http://localhost:3000/image.jpg" />
  );

  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should show the title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe("My Title");
  });

  it("should contain an image with url", () => {
    const img = wrapper.find("img");
    expect(img.props().src).toBe("http://localhost:3000/image.jpg");
    expect(img.prop("alt")).toBe("imagen");
  });

  it("should have animate__bounce", () => {
    const div = wrapper.find("div");
    expect(div.hasClass("animate__bounce")).toBe(true);
  });
});
