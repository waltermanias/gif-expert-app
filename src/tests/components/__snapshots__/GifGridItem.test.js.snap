// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem should render the component 1`] = `
<div
  className="card animate__animated animate__bounce animate__delay-2s"
>
  <img
    alt="imagen"
    src="http://localhost:3000/image.jpg"
  />
  <p>
    My Title
  </p>
</div>
`;
